### PR TITLE
Added Asphalt Events

### PR DESCRIPTION
**Pull request by DubStepMad**
---
Name: Asphalt Events
About: 
First commit, includes current website pages of Asphalt Events.
---

**Screenshots**
image

image

image

image

image

### CODE REVIEW

File: websites/A/AsphaltEvents/presence.ts [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+    };
+
+    presenceData.startTimestamp = browsingStamp;
+
+    if (document.location.hostname == "asphalt.events") {
+        if (document.location.pathname == "/") {
+            presenceData.details = "Browsing";
+            presenceData.state ="Upcoming Convoys & Events";
+      
+            delete presenceData.smallImageKey;
```
Comment (Bas950) [inline websites/A/AsphaltEvents/presence.ts:27]: Not needed since none is set.

File: websites/A/AsphaltEvents/presence.ts [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        } else if (document.location.pathname.includes("/event")) {
+            presenceData.details = "Viewing";
+            presenceData.state ="All Convoys & Events";
+      
+            presence.setActivity(presenceData);
+        } else if (document.location.pathname.includes("/profile")) {
+            presenceData.details = "Viewing";
+            presenceData.state ="Their Profile";
+      
+            delete presenceData.smallImageKey;
```
Comment (Bas950) [inline websites/A/AsphaltEvents/presence.ts:77]: Not needed since none is set.

File: websites/A/AsphaltEvents/presence.ts [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+            presence.setActivity(presenceData);
+        } else if (document.location.pathname.includes("/user/")) {
+            item = document.querySelector(
+                "body > div > main > div.container > div > div.row > div.col-md-9 > div > div > div > div > h1"
+              );
+
+            presenceData.details = "Viewing ";
+            presenceData.state = item.innerText+"'s Profile";
+      
+            delete presenceData.smallImageKey;
```
Comment (Bas950) [inline websites/A/AsphaltEvents/presence.ts:88]: Not needed since none is set.

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/AsphaltEvents/dist/metadata.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    "thumbnail": "https://asphalt.events/images/branding/Asphalt-Events-Trans.png",
+    "color": "#3490DC",
+    "tags": [
+      "ETS2",
+      "ATS",
+      "TruckersMP",
+      "Asphalt Events",
+      "Events",
+      "conovys"
+    ],
```
[the remark this answers is not in the document]
Comment (DubStepMad) [reply websites/A/AsphaltEvents/dist/metadata.json]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+
+var item: any,
+  user: any,
+  item2: any,
+  item3: any,
+  advert: any,
+  attendance: any,
+  output: any;
+
+var browsingStamp = Math.floor(Date.now() / 1000);
```
[the remark this answers is not in the document]
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+            presenceData.state ="Upcoming Convoys & Events";
+      
+            delete presenceData.smallImageKey;
+      
+            presence.setActivity(presenceData);
+        } else if (document.location.pathname.includes("/about")) {
+            presenceData.details = "Viewing";
+            presenceData.state ="About Us";
+      
+            delete presenceData.smallImageKey;
```
[the remark this answers is not in the document]
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts (hunk 3 above)
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts [hunk 7]
```diff
@@ -0,0 +1,128 @@
...
+            presenceData.state = item.innerText+"'s Profile";
+      
+            delete presenceData.smallImageKey;
+      
+            presence.setActivity(presenceData);
+        } else if (document.location.pathname.includes("/knowledge-base")) {
+            presenceData.details = "Browsing";
+            presenceData.state ="The Knowledge Base";
+
+            delete presenceData.smallImageKey;
```
[the remark this answers is not in the document]
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts [hunk 8]
```diff
@@ -0,0 +1,128 @@
...
+      
+            presence.setActivity(presenceData);
+        } else if (document.location.pathname.includes("/languages")) {
+            presenceData.details = "Translating The Website";
+            delete presenceData.state;
+
```
[the remark this answers is not in the document]
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts:82]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts [hunk 9]
```diff
@@ -0,0 +1,128 @@
...
+              );
+
+
+              if(item != null){
+                presenceData.state = item.innerText;
+              }else{
+                presenceData.details = "Translation Center";
+              }
+      
+            delete presenceData.smallImageKey;
```
[the remark this answers is not in the document]
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts (hunk 1 above)
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts [hunk 10]
```diff
@@ -0,0 +1,128 @@
...
+            presenceData.state ="About Us";
+      
+            delete presenceData.smallImageKey;
+      
+            presence.setActivity(presenceData);
+        } else if (document.location.pathname.includes("/event/create")) {
+            presenceData.details = "Creating";
+            presenceData.state ="Convoy or Event";
+      
+            delete presenceData.smallImageKey;
```
[the remark this answers is not in the document]
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+            if(item == "xClose" || item == null){
+                item = document.querySelector(
+                    "body > div > main > div.container > div > div.col-md-8 > div.card-header > h3"
+                  );
+            }
+
+            presenceData.details = "Viewing "+firstWord+':';
+            presenceData.state = item.innerText;
+      
+            delete presenceData.smallImageKey;
```
[the remark this answers is not in the document]
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts (hunk 2 above)
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+            item = document.querySelector(
+              "body > div > main > div.container > div > div.col-md-8 > div > div > h1 > strong"
+            );
+
+            item2 = document.querySelector(
+                "body > div > main > div.container > div > div.col-md-8 > div.card > div > h3"
+            );
+
+            var type = item2.innerText;
+            var firstWord = type.replace(/ .*/,'');
```
[the remark this answers is not in the document]
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/dist/metadata.json [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+    "thumbnail": "https://imgur.com/a/F5Ega8b",
+    "color": "#3490DC",
+    "tags": [
+      "ETS2",
+      "ATS",
+      "TruckersMP",
+      "AsphaltEvents",
+      "Events",
+      "convoys"
+    ],
```
Comment (Bas950) [inline websites/A/AsphaltEvents/dist/metadata.json:30]: tags cannot have cap letters

File: websites/A/AsphaltEvents/dist/metadata.json [hunk 14]
```diff
@@ -0,0 +1,32 @@
...
+      "asphalt.events",
+      "asphaltevents.com"
+    ],
+    "description": {
+      "en": "Modern and feature-packed convoy/event advertising service for TruckersMP."
+    },
+    "service": "Asphalt Events",
+    "version": "1.0.0",
+    "logo": "https://imgur.com/a/F5Ega8b",
+    "thumbnail": "https://imgur.com/a/F5Ega8b",
```
Comment (Bas950) [inline websites/A/AsphaltEvents/dist/metadata.json:21]: use the i.imgur links

File: websites/A/AsphaltEvents/presence.ts [hunk 15]
```diff
@@ -0,0 +1,105 @@
+const presence = new Presence({
+    clientId: "751816190653104168"
+});
+
+let item,
+    item2;
+
+const browsingStamp = Math.floor(Date.now() / 1000);
```
Comment (Bas950) [inline websites/A/AsphaltEvents/presence.ts:8]: move it above the item ones and merge the consts

File: websites/A/AsphaltEvents/presence.ts [hunk 16]
```diff
@@ -0,0 +1,105 @@
...
+            item = document.querySelector(
+                "body > div > form > div > div > div > div:nth-child(2) > select > option:checked"
+              ) as HTMLElement;
+
+
+              if(item != null){
+                presenceData.state = item.innerText;
+              }else{
+                presenceData.details = "Translation Center";
+              }
```
Comment (Bas950) [inline websites/A/AsphaltEvents/presence.ts:91]: shouldnt this be .state?

File: websites/A/AsphaltEvents/presence.ts [hunk 17]
```diff
@@ -0,0 +1,105 @@
...
+            presence.setActivity(presenceData);
+        } else if (document.location.pathname.includes("/knowledge-base")) {
+            presenceData.details = "Browsing";
+            presenceData.state ="The Knowledge Base";
+      
+            presence.setActivity(presenceData);
+        } else if (document.location.pathname.includes("/languages")) {
+            presenceData.details = "Translating The Website";
+            delete presenceData.state;
+
```
Comment (Bas950) [inline websites/A/AsphaltEvents/presence.ts:82]: not needed.

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/AsphaltEvents/dist/metadata.json (hunk 14 above)
Comment (DubStepMad) [reply websites/A/AsphaltEvents/dist/metadata.json]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/dist/metadata.json (hunk 13 above)
Comment (DubStepMad) [reply websites/A/AsphaltEvents/dist/metadata.json]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts (hunk 15 above)
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts (hunk 17 above)
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/presence.ts (hunk 16 above)
Comment (DubStepMad) [reply websites/A/AsphaltEvents/presence.ts]: Resolved in updated pull request. (Thanks for the point out)

File: websites/A/AsphaltEvents/dist/metadata.json [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+      "asphaltevents.com"
+    ],
+    "description": {
+      "en": "Modern and feature-packed convoy/event advertising service for TruckersMP."
+    },
+    "service": "Asphalt Events",
+    "version": "1.0.0",
+    "logo": "https://i.imgur.com/MXhf8fA.png",
+    "thumbnail": "https://i.imgur.com/MXhf8fA.png",
+    "color": "#3490DC",
```
Comment (Bas950) [inline websites/A/AsphaltEvents/dist/metadata.json:22]: thumbnail should be a promotional banner of the website, or a screenshot of the website, not the logo.

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/AsphaltEvents/dist/metadata.json (hunk 18 above)
Comment (DubStepMad) [reply websites/A/AsphaltEvents/dist/metadata.json:22]: Ah thank you, will resolve that now.

Comment (DubStepMad) [reply websites/A/AsphaltEvents/dist/metadata.json]: Resolved in updated pull request.

File: websites/A/AsphaltEvents/dist/metadata.json [hunk 19]
```diff
@@ -0,0 +1,32 @@
+{
+    "$schema": "https://schemas.premid.app/metadata/1.0",
+    "author": {
+      "name": "Senka Wolf",
+      "id": "170854083840704512"
+    },
+    "contributors": {
+      "name": "DubStepMad",
+      "id": "118810952530395136"
+    },
```
Comment (Bas950) [inline websites/A/AsphaltEvents/dist/metadata.json:10]: contributor needs to be in an array `"contributors": [{object}],`

**Review by Bas950 — CHANGES_REQUESTED** [review]

File: websites/A/AsphaltEvents/dist/metadata.json (hunk 19 above)
Comment (Bas950) [reply websites/A/AsphaltEvents/dist/metadata.json:10]: _Should be your last needed change_

Comment (DubStepMad) [reply websites/A/AsphaltEvents/dist/metadata.json]: Resolved in updated pull request.

**Review by Bas950 — APPROVED** [review]

**Review by Alanexei — APPROVED** [review]

**Pull request merged**